ytypes/leaf.go

```diff
@@ -71,13 +72,17 @@ func validateLeaf(inSchema *yang.Entry, value interface{}) util.Errors {
 			return util.NewErrs(fmt.Errorf("bad leaf type: expect Bool for empty type for schema %s, have type %v", schema.Name, ykind))
 		}
 		rv = value
+	case reflect.Int8, reflect.Int16, reflect.Int32, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64, reflect.Float64, reflect.String:
+		if ykind != yang.Yunion {
+			return util.NewErrs(fmt.Errorf("bad leaf type: expect %v for union type for schema %s, have type %v", rkind, schema.Name, ykind))
+		}
```

Comment:
why do the integer cases (both Int64 above and these cases) not assign "rv = value"?

Reply:
Nice find, doing this actually breaks, and the reason is because of some unnecessary complexity in the existing code. This is fixed now.